soco/core.py

```diff
@@ -1578,6 +1580,246 @@ def _build_album_art_full_uri(self, url):
     def _music_lib_search(self, search, start, max_items):
         return self.music_library._music_lib_search(search, start, max_items)
 
+    @only_on_master
+    def reorder_sonos_playlist(self, sonos_playlist, tracks, new_pos,
+                               update_id=0):
+        """Reorder and/or Remove tracks in a Sonos playlist.
+
+        The underlying call is quite complex as it both can move a track within
+        the list or delete a track from the playlist.  All of this depends on
+        what tracks and new_pos specify.
+
+        If a list is specified for tracks, then a list must be used for
+        new_pos. Each list element is a discrete modification and the next
+        list operation must anticipate the new state of the playlist.
+
+        If a comma formatted string to tracks is specified, then use
+        a similiar string to specify new_pos. Those operations should be
+        ordered from the end of the list to the beginning
+
+        See the helper methods
+        :py:meth:`clear_sonos_playlist`, :py:meth:`move_in_sonos_playlist`,
+        :py:meth:`remove_from_sonos_playlist` for simplified usage.
+
+        update_id - If you have a series of operations, tracking the update_id
+        and setting it, will save a lookup operation.
+
+        Examples:
+          To reorder the first two tracks::
+
+            # sonos_playlist specified by the DidlPlaylistContainer object
+            sonos_playlist = device.get_sonos_playlists()[0]
+            device.reorder_sonos_playlist(sonos_playlist,
+                                          tracks=[0, ], new_pos=[1, ])
```

Comment:
Commas make lists unambiguous.  While not required for a single integer,
['this',] != ['this'].  Any iterable will be split into elements within the
list if you don't mark them with commas.  If you would like me to remove
them, I will.  Personally, I don't like to vary based on data types, so I
choose to be explicit.   If you have a pep I'd be happy to read and perhaps
alter my opinion, but I've not read any thing authoritative against it.
 "Explicit is better than implicit." -
https://www.python.org/dev/peps/pep-0020/

On Mon, Sep 14, 2015 at 7:33 AM, Kenneth Nielsen <notifications@github.com>
wrote:

> In soco/core.py
> <https://github.com/SoCo/SoCo/pull/352#discussion_r39387778>:
>
> > +        ordered from the end of the list to the beginning
> > +
> > +        See the helper methods
> > +        :py:meth:`clear_sonos_playlist`, :py:meth:`move_in_sonos_playlist`,
> > +        :py:meth:`remove_from_sonos_playlist` for simplified usage.
> > +
> > +        update_id - If you have a series of operations, tracking the update_id
> > +        and setting it, will save a lookup operation.
> > +
> > +        Examples:
> > +          To reorder the first two tracks::
> > +
> > +            # sonos_playlist specified by the DidlPlaylistContainer object
> > +            sonos_playlist = device.get_sonos_playlists()[0]
> > +            device.reorder_sonos_playlist(sonos_playlist,
> > +                                          tracks=[0, ], new_pos=[1, ])
>
> On the really nit-picky end of things, these commas in [0, ] are not
> necessary when dealing with lists the way they are for tuples. It is
> considered good style to add these trailing commas for *multi-line* lists
> (because it will make modifications easier to read in the version control
> diffs), but I believe not for single line lists.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SoCo/SoCo/pull/352/files#r39387778>.
>



-- 
Best,

Jeff Hinrichs
dundeemt@gmail.com
402.218.1473